authentication/views.py

```diff
@@ -27,16 +27,33 @@ def post(self, request):
         if form.is_valid():
 
             form_data = form.clean()
+
             new_user_data = {
                     'username': form_data.get('username'),
                     'email': form_data.get('email'),
                     'password': form_data.get('password')
             }
+
+            user_other_fields = {
+                'first_name': form_data.get('first_name'),
+                'last_name': form_data.get('last_name'),
+            }
             try:
-                User.objects.create_user(new_user_data['username'], new_user_data['email'], new_user_data['password'])
+                email = form.email_clean(form_data.get('email'))
```

Comment:
instead putting validation logic here it should be handled in form validations
https://docs.djangoproject.com/en/2.2/ref/forms/validation/